### PR TITLE
Replace `sticker-*` by a single `sticker` command

### PR DESCRIPTION
**Pull request by danieldk**
- The old utilities are now subcommands of sticker.
- sticker has a `completions` subcommand that generates completion
  definitions for various shells.
- `sticker` and `sticker --version` report the crate's version.
- The various applications are made more uniform through the
  `StickerPipelineApp` and `StickerTrainApp` traits.

### CODE REVIEW

**Review by sebpuetz — APPROVED** [review]
Looks good to me, although it's impossible to catch everything with 1k+ moved/changed lines.

File: sticker-utils/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            subcommands::ServerApp::parse(matches.subcommand_matches("server").unwrap()).run()
+        }
+        "tag" => subcommands::TagApp::parse(matches.subcommand_matches("tag").unwrap()).run(),
+        "train" => subcommands::TrainApp::parse(matches.subcommand_matches("train").unwrap()).run(),
+        _unknown => unreachable!(),
+    }
+}
+
+fn write_completion_script(mut cli: App, shell: Shell) {
+    cli.gen_completions_to("finalfusion", shell, &mut stdout());
```
Comment (twuebi) [inline sticker-utils/src/main.rs]:
```sed
s/finalfusion/sticker/
```

**Comment by danieldk** [comment]
> Looks good to me, although it's impossible to catch everything with 1k+ moved/changed lines.

Yep. I did try all the subcommands this time ;).

File: sticker-utils/src/main.rs (hunk 1 above)
Comment (danieldk) [reply sticker-utils/src/main.rs]: Good catch!

Comment (danieldk) [reply sticker-utils/src/main.rs]: fixed

**Pull request merged**